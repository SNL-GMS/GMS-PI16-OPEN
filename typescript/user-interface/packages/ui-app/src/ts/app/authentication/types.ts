import { UserSessionTypes } from '@gms/ui-state';

export interface Authenticator {
  authenticateWith(userName: string): Promise<UserSessionTypes.AuthStatus>;
  checkIsAuthenticated(): Promise<UserSessionTypes.AuthStatus>;
  unAuthenticateWith(): Promise<UserSessionTypes.AuthStatus>;
  logout(setAuthStatus: (status: UserSessionTypes.AuthStatus) => void): void;
}

export interface LoginRequestParams {
  userName: string;
}
