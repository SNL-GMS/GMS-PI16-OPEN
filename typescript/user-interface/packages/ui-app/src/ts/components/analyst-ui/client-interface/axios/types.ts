/* eslint-disable @typescript-eslint/no-explicit-any */
import {
  ChannelSegmentTypes,
  EventTypes,
  QcMaskTypes,
  SignalDetectionTypes,
  WaveformTypes,
  WorkflowTypes
} from '@gms/common-model';
import { AxiosRequestConfig } from 'axios';
import { QueryResult } from 'react-query';

export interface EventMutationArgs {
  variables: {
    events: EventTypes.Event[];
  };
}

export interface QcMaskMutationArgs {
  variables: {
    qcMasks: QcMaskTypes.QcMask[];
  };
}

export interface SignalDetectionMutationArgs {
  variables: {
    signalDetections: SignalDetectionTypes.SignalDetection[];
  };
}

// ***************************************
// Workflow Interval Query and Mutation interfaces
// ***************************************
export interface StageIntervalsByStageIdAndTimeRequest {
  startTime: number;
  endTime: number;
  stageIds: WorkflowTypes.WorkflowDefinitionId[];
}
export interface UserRequest {
  userName: string;
  time: number;
}

export interface UpdateActivityIntervalStatusRequest extends UserRequest {
  activityIntervalId: WorkflowTypes.IntervalId;
  stageIntervalId: WorkflowTypes.IntervalId;
  status: WorkflowTypes.IntervalStatus;
}

export interface UpdateInteractiveAnalysisStageIntervalStatusRequest extends UserRequest {
  stageIntervalId: WorkflowTypes.IntervalId;
  status: WorkflowTypes.IntervalStatus;
}

export interface SplitActivityIntervalRequest {
  intervalId: WorkflowTypes.IntervalId;
  duration: number;
}
export interface EventQueryProps {
  eventQuery: QueryResult<EventTypes.Event[], any>;
}

export interface QcMaskQueryProps {
  qcMaskQuery: QueryResult<QcMaskTypes.QcMask[], any>;
}

export interface SignalDetectionQueryProps {
  signalDetectionQuery: QueryResult<SignalDetectionTypes.SignalDetection[], any>;
}

export interface WaveformQueryProps {
  waveformQuery: QueryResult<ChannelSegmentTypes.ChannelSegment<WaveformTypes.Waveform>[], any>;
}

/**
 * An axios request known to have a waveform query request in its data.
 */
export type WaveformAxiosRequestConfig = AxiosRequestConfig & {
  data: WaveformTypes.WaveformQueryArgs;
};
