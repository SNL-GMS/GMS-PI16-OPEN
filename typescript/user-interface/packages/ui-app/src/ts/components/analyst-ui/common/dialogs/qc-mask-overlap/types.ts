import { QcMaskTypes } from '@gms/common-model';

// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { CoordinatesPx } from '../qc-mask-form/types';
// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { QcMaskDialogBoxType } from '../types';

/**
 * QcMaskOverlap Props
 */
export interface QcMaskOverlapProps {
  masks: QcMaskTypes.QcMask[];
  // If set, any context menus opened as children will be created at given x/y coordinates
  contextMenuCoordinates?: CoordinatesPx;
  openNewContextMenu(
    x: number,
    y: number,
    mask: QcMaskTypes.QcMask,
    dialogType: QcMaskDialogBoxType
  );
  selectMask(mask: QcMaskTypes.QcMask);
}

/**
 * QcMaskOverlap  State
 */
export interface QcMaskOverlapState {
  selectedMask: QcMaskTypes.QcMask;
  createDialogOfType?: QcMaskDialogBoxType;
}
export interface QcMaskTableButtonParams {
  onClick(x: number, y: number, params: any);
}
