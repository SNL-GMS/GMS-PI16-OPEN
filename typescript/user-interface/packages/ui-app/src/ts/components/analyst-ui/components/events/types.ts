import { GQLTypes } from '@gms/common-graphql';
import { CommonTypes, EventTypes, WorkflowTypes } from '@gms/common-model';
import GoldenLayout from '@gms/golden-layout';
import { ChildProps, MutationFunction } from 'react-apollo';

/**
 * Types of events which can be show
 */
export enum EventFilters {
  EDGE = 'Edge',
  COMPLETED = 'Completed'
}

/**
 * Table row object for events
 */
export interface EventsRow {
  id: string;
  eventHypId: string;
  isOpen: boolean;
  stageId: string;
  lat: number;
  lon: number;
  depth: number;
  time: number;
  activeAnalysts: string[];
  numDetections: number;
  status: string;
  edgeEvent: boolean;
  signalDetectionConflicts: SignalDetectionConflict[];
}

export interface SignalDetectionConflict {
  stationName: string;
  id: string;
  arrivalTime: number;
  phase: CommonTypes.PhaseType;
}
/**
 * Event list local state
 */
export interface EventsState {
  currentTimeInterval: CommonTypes.TimeRange;
  suppressScrollOnNewData: boolean;
  showEventOfType: Map<EventFilters, boolean>;
}

/**
 * Mutations used in the event list
 */
export interface EventsMutations {
  updateEvents: MutationFunction;
  saveEvent: MutationFunction;
  saveAllModifiedEvents: MutationFunction;
}

/**
 * Props mapped in from Redux state
 */
export interface EventsReduxProps {
  // Passed in from golden-layout
  glContainer?: GoldenLayout.Container;
  currentTimeInterval: CommonTypes.TimeRange;
  analysisMode: WorkflowTypes.AnalysisMode;
  openEventId: string;
  selectedEventIds: string[];

  // callbacks
  setOpenEventId(
    event: EventTypes.Event | undefined,
    latestLocationSolutionSet: EventTypes.LocationSolutionSet | undefined,
    preferredLocationSolutionId: string | undefined
  ): void;
  setSelectedEventIds(ids: string[]): void;
}

/**
 * Consolidated props type for event list
 */
export type EventsProps = EventsReduxProps &
  ChildProps<EventsMutations> &
  GQLTypes.EventsInTimeRangeQueryProps &
  GQLTypes.DefaultStationsQueryProps &
  GQLTypes.SignalDetectionsByStationQueryProps &
  GQLTypes.WorkspaceStateProps;
