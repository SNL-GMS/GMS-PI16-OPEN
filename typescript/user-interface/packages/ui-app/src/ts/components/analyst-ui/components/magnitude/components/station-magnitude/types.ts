import { EventTypes } from '@gms/common-model';
import { AnalystWorkspaceTypes } from '@gms/ui-state';
import Immutable from 'immutable';
import { MutationFunction } from 'react-apollo';

// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { AmplitudesByStation } from '../../types';

/**
 * StationMagnitude State
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface StationMagnitudeState {
  computeNetworkMagnitudeSolutionStatus: Immutable.Map<
    string,
    [{ stationName: string; rational: string }]
  >;
}

export interface MagnitudeAndSdData {
  magSolution: EventTypes.NetworkMagnitudeSolution;
  sdData: StationMagnitudeSdData;
}

/**
 * options that can be passed to ag grid
 */
export interface Options {
  alignedGrids: any[];
}

/**
 * StationMagnitude Props
 */
export interface StationMagnitudeProps {
  options?: Options;
  amplitudesByStation: AmplitudesByStation[];
  historicalMode: boolean;
  selectedSdIds: string[];
  locationSolution: EventTypes.LocationSolution;
  displayedMagnitudeTypes: AnalystWorkspaceTypes.DisplayedMagnitudeTypes;
  computeNetworkMagnitudeSolution: MutationFunction;
  openEventId: string;

  setSelectedSdIds(ids: string[]): void;
  checkBoxCallback(
    magnitudeType: EventTypes.MagnitudeType,
    stationNames: string[],
    defining: boolean
  ): Promise<[{ stationName: string; rational: string }]>;
}

export interface StationMagnitudeSdData {
  channel: string;
  phase: string;
  amplitudeValue: number;
  amplitudePeriod: number;
  signalDetectionId: string;
  time: number;
  stationName: string;
  flagForReview: boolean;
}

export interface MagnitudeDataForRow {
  channel: string;
  signalDetectionId: string;
  phase: string;
  amplitudeValue: number;
  amplitudePeriod: number;
  flagForReview: boolean;
  defining: boolean;
  mag: number;
  res: number;
  hasMagnitudeCalculationError: boolean;
  computeNetworkMagnitudeSolutionStatus: string;
}

/**
 * Table row object for station magnitude
 */
export interface StationMagnitudeRow {
  id: string;
  dataForMagnitude: Map<EventTypes.MagnitudeType, MagnitudeDataForRow>;
  station: string;
  dist: number;
  azimuth: number;
  selectedSdIds: string[];
  historicalMode: boolean;
  azimuthTooltip: string;
  checkBoxCallback(
    magnitudeType: EventTypes.MagnitudeType,
    stationNames: string[],
    defining: boolean
  ): Promise<void>;
}
export interface StationMagAndSignalDetection {
  stationName: string;
  magnitudeAndSdData: Map<EventTypes.MagnitudeType, MagnitudeAndSdData>;
}
