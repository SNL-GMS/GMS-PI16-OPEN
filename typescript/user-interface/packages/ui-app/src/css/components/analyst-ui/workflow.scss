@import '../../gms-colors.scss';
@import '../../gms-fonts.scss';

$BLOCK_HEIGHT_PX: 24px;
$ROW_COMPUTED_HEIGHT_PX: 26px;
$BLOCK_WIDTH_PX: 140px;
$TABLE_MARGIN: 34px;
$DIVIDER_MARGIN: 31px;
//The table margin plus some manual fiddling
$WIDTH_OF_LEFT_CURTAIN: 43px;
$gap-between-rows: 0.75em;

.gms-workflow-time-axis {
  path,
  line {
    fill: none;
    stroke: $gms-prominent;
    shape-rendering: crispEdges;
  }
  text {
    font-family: $gms-sans;
    font-size: 12px;
    fill: $gms-main;
    &.day-label {
      stroke: $workflow-day-divider;
      font-size: 13px;
    }
  }
}

.workflow-display-window {
  height: 100%;
}

.workflow-panel {
  height: 100%;
  width: 100%;
}

.workflow-scroll-wrapper {
  max-height: calc(100% - 55px);
  overflow-y: scroll;
}

.interval-table-wrapper {
  flex: 1 0 auto;
  display: flex;
  margin-left: $TABLE_MARGIN;
  margin-right: $BLOCK_WIDTH_PX;
  flex-direction: column;
  overflow-x: scroll;
  overflow-y: hidden;
  position: static;
  gap: 2px;
}

.interval-table-wrapper-wrapper {
  position: relative;
  overflow-y: hidden;
  overflow-x: hidden;
}

// Left is set to account for the workflow-table-wrapper's left margin, minus the size of the button
.stage-row__expand-button {
  height: $BLOCK_HEIGHT_PX;
  border: 1px solid $gms-recessed;
  width: $BLOCK_HEIGHT_PX;
  transition: margin-bottom 0.15s;
  min-height: $ROW_COMPUTED_HEIGHT_PX;
  min-width: $ROW_COMPUTED_HEIGHT_PX;
  padding: 0;
  position: absolute;
  left: 7px;
  margin-top: -1px;
  z-index: 5;
}

.stage-row {
  display: flex;
  gap: 2px;
}

.stage-column {
  display: flex;
  flex-direction: column;
  gap: 2px;
}

.internal-row {
  display: grid;
  grid-auto-columns: 8px;
  gap: 2px;
}

.workflow-tooltip-container {
  display: grid;
  grid-template-columns: 2fr 3fr;
  margin: 0 auto;
  gap: 0.25em;
  padding-bottom: 0.5em;
  border-bottom: 1px dashed $gms-soft;
}

.workflow-tooltip-container .label-value__label {
  font-weight: bold;
}

.workflow-tooltip-container:last-child {
  border-bottom: none;
  padding-bottom: 0;
}

.workflow-tooltip-container + .workflow-tooltip-container {
  padding-top: 0.5em;
}

// The absolute position lets it float to the left over the workflow display
// The margin right and top are for alignment
.workflow-table-label {
  height: $ROW_COMPUTED_HEIGHT_PX;
  width: $BLOCK_WIDTH_PX;
  border: 1px solid;
  display: flex;
  border-color: $gms-background;
  justify-content: center;
  align-items: center;
  background: $gms-soft;
  flex: 0 0 auto;
  position: absolute;
  right: 0px;
  margin-right: 4px;
  margin-top: -1px;
  z-index: 5;
}

.workflow-table-label--activity {
  filter: $interval-activity-darken-cell-by;
}

.workflow-row {
  position: relative;
  display: flex;
  margin-bottom: 2px;
}

.interval-cell {
  display: flex;
  flex-direction: column;
  height: 100%;
  height: $BLOCK_HEIGHT_PX;
  border: 1px solid $gms-background;
  align-items: center;
  justify-content: center;
  margin: 0;
  gap: 2px;
  overflow: hidden;
  white-space: nowrap;
  cursor: pointer;
  .percent-bar {
    height: 100%;
    width: 100%;
  }
}

.interval-cell:hover {
  filter: brightness(1.2);
}

.interval-cell--in-progress {
  background-color: $interval-in-progress-fill;
  border: 1px solid $interval-in-progress-border;
  .percent-bar {
    background-color: scale-color(
      $color: $interval-in-progress-fill,
      $lightness: 25%,
      $saturation: -15%
    );
  }
}

.interval-cell--complete {
  background-color: $interval-complete-fill;
  border: 1px solid $interval-complete-border;
}

.interval-cell--not-complete {
  background-color: $interval-not-complete-fill;
  border: 1px solid $interval-not-complete-border;
}

.interval-cell--not-started {
  background-color: $interval-not-started-fill;
  border: 1px solid $interval-not-started-border;
}

.interval-cell--skipped {
  background-color: $interval-skipped-fill;
  border: 1px solid $interval-skipped-border;
}

.interval-cell--failed {
  background-color: $interval-failed-fill;
  border: 1px solid $interval-failed-border;
}

.interval-cell--selected {
  border: 2px solid $gms-selection;
}

.interval-cell--activity-cell {
  filter: $interval-activity-darken-cell-by;
}

.interval-cell--activity-cell:hover {
  filter: brightness(1);
}

.interval-cell--stale {
  filter: brightness(0.5);
}

.workflow-day-divider {
  position: absolute;
  top: 0px;
  bottom: 0px;
  width: 4px;
  z-index: 1;
  margin-left: $DIVIDER_MARGIN;
}

// To account for the offset of the workflow-table, the time axis needs to be moved to left by 1
.time-axis-wrapper {
  margin-left: -1px;
  width: calc(100%-131px);
  height: 25px;
  flex: 0 0 auto;
  position: static;
  z-index: 1;
}

.time-axis {
  width: 100%;
  height: 25px;
  overflow: hidden;
}

.interval-table__curtain-left {
  width: $WIDTH_OF_LEFT_CURTAIN;
  position: absolute;
  background-color: $gms-background;
  height: 100vh;
  z-index: 2;
  left: -6px;
}

.workflow-ellipsis {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: 100px;
  text-align: center;
}

.workflow-open-anything-row {
  display: flex;
  flex-flow: wrap;
  box-sizing: border-box;
  align-items: center;
  min-width: 56ch;
  & > * {
    margin-left: 0.5rem;
  }
}

.open-anything-contents .dropdown-container {
  display: flex;
  justify-content: space-between;
}

.workflow-open-anything-row .dropdown-container .span {
  display: flex;
}

.open-anything-contents .date-range-picker {
  width: 100%;
  display: flex;
  margin-bottom: 5px;
}

.open-anything-contents .date-range-picker .date-range-picker__range-input {
  width: 65%;
  margin: 0px;
}

.open-anything-contents
  .date-range-picker
  .date-range-picker__range-input
  .bp3-popover-target
  .bp3-control-group
  * {
  flex-shrink: 1;
}

.open-anything-contents .date-range-picker .date-range-picker__label {
  font-size: 1.125em;
  font-weight: 600;
  margin: 0px;
  width: 35%;
}

.open-anything-contents .dropdown-label {
  font-size: 1.125em;
  font-weight: 600;
  margin: 0px;
  width: 35%;
}

.open-anything-contents .dropdown-selector {
  margin-bottom: 5px;
  width: 65%;
}

.interval-confirmation-contents {
  display: flex;
}

.interval-confirmation-text {
  width: 90%;
}

.interval-confirmation-icon {
  width: 10%;
}

.interval-confirmation-header {
  font-weight: bold;
}

.open-anything-error {
  color: $gms-strong-warning;
  display: flex;
  min-height: 20px;
}
.open-anything-error-text {
  margin-left: 5px;
  width: calc(100%-20px);
}
