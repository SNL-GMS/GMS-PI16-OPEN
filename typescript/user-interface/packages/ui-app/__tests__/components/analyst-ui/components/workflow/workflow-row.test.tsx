/* eslint-disable react/jsx-props-no-spreading */
import { createStore } from '@gms/ui-state';
import * as Enzyme from 'enzyme';
import * as React from 'react';
import { Provider } from 'react-redux';

import { WorkflowRowProps } from '../../../../../src/ts/components/analyst-ui/components/workflow/types';
import { WorkflowContext } from '../../../../../src/ts/components/analyst-ui/components/workflow/workflow-context';
import { WorkflowRow } from '../../../../../src/ts/components/analyst-ui/components/workflow/workflow-row';
import * as WorkflowDataTypes from './workflow-data-types';

// set up window alert and open so we don't see errors
window.alert = jest.fn();
window.open = jest.fn();

window.ResizeObserver = jest.fn(() => {
  return { observe: jest.fn(), disconnect: jest.fn(), unobserve: jest.fn() };
});
const globalAny: any = global;
globalAny.ResizeObserver = window.ResizeObserver;
globalAny.DOMRect = jest.fn(() => ({}));

const MOCK_TIME = 1606818240000;
global.Date.now = jest.fn(() => MOCK_TIME);

const store: any = createStore();

describe('Workflow Row', () => {
  it('is exported', () => {
    expect(WorkflowRow).toBeDefined();
  });

  const workflowRowProps: WorkflowRowProps = {
    stageIntervals: [WorkflowDataTypes.interactiveAnalysisStageInterval],
    subRowNames: []
  };

  it('matches snapshot', () => {
    const component = Enzyme.mount(
      <Provider store={store}>
        <WorkflowContext.Provider
          value={{
            staleStartTime: 1,
            allActivitiesOpenForSelectedInterval: false,
            openConfirmationPrompt: jest.fn(),
            openAnythingConfirmationPrompt: jest.fn(),
            closeConfirmationPrompt: jest.fn()
          }}
        >
          <WorkflowRow {...workflowRowProps} />
        </WorkflowContext.Provider>
      </Provider>
    );
    expect(component).toMatchSnapshot();
    const workflowRow: any = component.find('WorkflowRow').instance();
    expect(workflowRow).toBeDefined();
  });

  it('shallow mounts', () => {
    const shallow = Enzyme.shallow(
      <Provider store={store}>
        <WorkflowRow {...workflowRowProps} />
      </Provider>
    );
    expect(shallow).toMatchSnapshot();
    const workflowRow: any = shallow.find('WorkflowRow');
    expect(workflowRow).toBeDefined();
  });
});
