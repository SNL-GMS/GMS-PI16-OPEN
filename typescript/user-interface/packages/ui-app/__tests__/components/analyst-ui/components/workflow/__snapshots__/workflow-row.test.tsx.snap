// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workflow Row matches snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <WorkflowRow
    stageIntervals={
      Array [
        Object {
          "activityIntervals": Array [
            Object {
              "activeAnalysts": Array [
                "larry",
                "moe",
                "curly",
              ],
              "comment": "interval example",
              "endTime": 1622057187,
              "modificationTime": 1622057667,
              "name": "Event Review",
              "percentAvailable": 100,
              "processingEndTime": 1622057547,
              "processingStartTime": 1622057487,
              "stageName": "AL1",
              "startTime": 1622053587,
              "status": "IN_PROGRESS",
              "storageTime": 1622057607,
            },
          ],
          "comment": "interval example",
          "endTime": 1622057187,
          "modificationTime": 1622057667,
          "name": "AL1",
          "percentAvailable": 100,
          "processingEndTime": 1622057547,
          "processingStartTime": 1622057487,
          "stageMetrics": Object {
            "associatedSignalDetectionCount": 34,
            "eventCount": 21,
            "maxMagnitude": 8,
            "unassociatedSignalDetectionCount": 55,
          },
          "stageMode": "INTERACTIVE",
          "startTime": 1622053587,
          "status": "IN_PROGRESS",
          "storageTime": 1622057607,
        },
      ]
    }
    subRowNames={Array []}
  >
    <div
      className="stage-row"
    >
      <Memo()
        disabled={false}
        isExpanded={false}
        stageName="AL1"
        toggle={[Function]}
      >
        <Blueprint3.Button
          className="stage-row__expand-button"
          disabled={false}
          icon="small-plus"
          key="AL1"
          onClick={[Function]}
        >
          <button
            className="bp3-button stage-row__expand-button"
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            type="button"
          >
            <Blueprint3.Icon
              icon="small-plus"
              key="leftIcon"
            >
              <span
                className="bp3-icon bp3-icon-small-plus"
                icon="small-plus"
              >
                <svg
                  data-icon="small-plus"
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <desc>
                    small-plus
                  </desc>
                  <path
                    d="M11 7H9V5c0-.55-.45-1-1-1s-1 .45-1 1v2H5c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1V9h2c.55 0 1-.45 1-1s-.45-1-1-1z"
                    fillRule="evenodd"
                    key="0"
                  />
                </svg>
              </span>
            </Blueprint3.Icon>
            <Blueprint3.Icon
              key="rightIcon"
            />
          </button>
        </Blueprint3.Button>
      </Memo()>
      <WorkflowTableStage
        isExpanded={false}
        key="1622053587 AL1"
        stage={
          Object {
            "activityIntervals": Array [
              Object {
                "activeAnalysts": Array [
                  "larry",
                  "moe",
                  "curly",
                ],
                "comment": "interval example",
                "endTime": 1622057187,
                "modificationTime": 1622057667,
                "name": "Event Review",
                "percentAvailable": 100,
                "processingEndTime": 1622057547,
                "processingStartTime": 1622057487,
                "stageName": "AL1",
                "startTime": 1622053587,
                "status": "IN_PROGRESS",
                "storageTime": 1622057607,
              },
            ],
            "comment": "interval example",
            "endTime": 1622057187,
            "modificationTime": 1622057667,
            "name": "AL1",
            "percentAvailable": 100,
            "processingEndTime": 1622057547,
            "processingStartTime": 1622057487,
            "stageMetrics": Object {
              "associatedSignalDetectionCount": 34,
              "eventCount": 21,
              "maxMagnitude": 8,
              "unassociatedSignalDetectionCount": 55,
            },
            "stageMode": "INTERACTIVE",
            "startTime": 1622053587,
            "status": "IN_PROGRESS",
            "storageTime": 1622057607,
          }
        }
        subRowNames={Array []}
      >
        <div
          className="stage-column"
        >
          <Tooltip
            activeAnalysts={
              Array [
                "larry",
                "moe",
                "curly",
              ]
            }
            interval={
              Object {
                "activityIntervals": Array [
                  Object {
                    "activeAnalysts": Array [
                      "larry",
                      "moe",
                      "curly",
                    ],
                    "comment": "interval example",
                    "endTime": 1622057187,
                    "modificationTime": 1622057667,
                    "name": "Event Review",
                    "percentAvailable": 100,
                    "processingEndTime": 1622057547,
                    "processingStartTime": 1622057487,
                    "stageName": "AL1",
                    "startTime": 1622053587,
                    "status": "IN_PROGRESS",
                    "storageTime": 1622057607,
                  },
                ],
                "comment": "interval example",
                "endTime": 1622057187,
                "modificationTime": 1622057667,
                "name": "AL1",
                "percentAvailable": 100,
                "processingEndTime": 1622057547,
                "processingStartTime": 1622057487,
                "stageMetrics": Object {
                  "associatedSignalDetectionCount": 34,
                  "eventCount": 21,
                  "maxMagnitude": 8,
                  "unassociatedSignalDetectionCount": 55,
                },
                "stageMode": "INTERACTIVE",
                "startTime": 1622053587,
                "status": "IN_PROGRESS",
                "storageTime": 1622057607,
              }
            }
            key="1622053587 AL1"
          >
            <TooltipWrapper
              content={
                <TooltipPanel
                  activeAnalysts="larry, moe, curly"
                  isStale={false}
                  lastModified="2021-05-26 19:34:27"
                  status="In Progress"
                />
              }
            >
              <Blueprint3.Tooltip
                boundary="window"
                className="core-tooltip"
                content={
                  <TooltipPanel
                    activeAnalysts="larry, moe, curly"
                    isStale={false}
                    lastModified="2021-05-26 19:34:27"
                    status="In Progress"
                  />
                }
                hoverCloseDelay={0}
                hoverOpenDelay={300}
                position="bottom"
                targetClassName="core-tooltip__target"
                targetTagName="div"
                transitionDuration={100}
                wrapperTagName="div"
              >
                <Blueprint3.Popover
                  autoFocus={false}
                  boundary="window"
                  canEscapeKeyClose={false}
                  captureDismiss={false}
                  className="core-tooltip"
                  content={
                    <TooltipPanel
                      activeAnalysts="larry, moe, curly"
                      isStale={false}
                      lastModified="2021-05-26 19:34:27"
                      status="In Progress"
                    />
                  }
                  defaultIsOpen={false}
                  disabled={false}
                  enforceFocus={false}
                  fill={false}
                  hasBackdrop={false}
                  hoverCloseDelay={0}
                  hoverOpenDelay={300}
                  inheritDarkTheme={true}
                  interactionKind="hover-target"
                  lazy={true}
                  minimal={false}
                  modifiers={Object {}}
                  openOnTargetFocus={true}
                  popoverClassName="bp3-tooltip"
                  position="bottom"
                  targetClassName="core-tooltip__target"
                  targetTagName="div"
                  transitionDuration={100}
                  usePortal={true}
                  wrapperTagName="div"
                >
                  <Manager>
                    <div
                      className="bp3-popover-wrapper core-tooltip"
                    >
                      <Reference
                        innerRef={[Function]}
                      >
                        <InnerReference
                          innerRef={[Function]}
                          setReferenceNode={[Function]}
                        >
                          <Blueprint3.ResizeSensor
                            onResize={[Function]}
                          >
                            <div
                              className="bp3-popover-target core-tooltip__target"
                              onBlur={[Function]}
                              onFocus={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                            >
                              <div
                                className="internal-row"
                                key=".0"
                                tabIndex={0}
                              >
                                <Memo()
                                  stageInterval={
                                    Object {
                                      "activityIntervals": Array [
                                        Object {
                                          "activeAnalysts": Array [
                                            "larry",
                                            "moe",
                                            "curly",
                                          ],
                                          "comment": "interval example",
                                          "endTime": 1622057187,
                                          "modificationTime": 1622057667,
                                          "name": "Event Review",
                                          "percentAvailable": 100,
                                          "processingEndTime": 1622057547,
                                          "processingStartTime": 1622057487,
                                          "stageName": "AL1",
                                          "startTime": 1622053587,
                                          "status": "IN_PROGRESS",
                                          "storageTime": 1622057607,
                                        },
                                      ],
                                      "comment": "interval example",
                                      "endTime": 1622057187,
                                      "modificationTime": 1622057667,
                                      "name": "AL1",
                                      "percentAvailable": 100,
                                      "processingEndTime": 1622057547,
                                      "processingStartTime": 1622057487,
                                      "stageMetrics": Object {
                                        "associatedSignalDetectionCount": 34,
                                        "eventCount": 21,
                                        "maxMagnitude": 8,
                                        "unassociatedSignalDetectionCount": 55,
                                      },
                                      "stageMode": "INTERACTIVE",
                                      "startTime": 1622053587,
                                      "status": "IN_PROGRESS",
                                      "storageTime": 1622057607,
                                    }
                                  }
                                >
                                  <Memo()
                                    key="AL1"
                                    stageInterval={
                                      Object {
                                        "activityIntervals": Array [
                                          Object {
                                            "activeAnalysts": Array [
                                              "larry",
                                              "moe",
                                              "curly",
                                            ],
                                            "comment": "interval example",
                                            "endTime": 1622057187,
                                            "modificationTime": 1622057667,
                                            "name": "Event Review",
                                            "percentAvailable": 100,
                                            "processingEndTime": 1622057547,
                                            "processingStartTime": 1622057487,
                                            "stageName": "AL1",
                                            "startTime": 1622053587,
                                            "status": "IN_PROGRESS",
                                            "storageTime": 1622057607,
                                          },
                                        ],
                                        "comment": "interval example",
                                        "endTime": 1622057187,
                                        "modificationTime": 1622057667,
                                        "name": "AL1",
                                        "percentAvailable": 100,
                                        "processingEndTime": 1622057547,
                                        "processingStartTime": 1622057487,
                                        "stageMetrics": Object {
                                          "associatedSignalDetectionCount": 34,
                                          "eventCount": 21,
                                          "maxMagnitude": 8,
                                          "unassociatedSignalDetectionCount": 55,
                                        },
                                        "stageMode": "INTERACTIVE",
                                        "startTime": 1622053587,
                                        "status": "IN_PROGRESS",
                                        "storageTime": 1622057607,
                                      }
                                    }
                                  >
                                    <div
                                      className="interval-cell interval-cell--in-progress"
                                      key="1622053587"
                                      onContextMenu={[Function]}
                                      style={
                                        Object {
                                          "gridColumn": "span 12",
                                        }
                                      }
                                    />
                                  </Memo()>
                                </Memo()>
                              </div>
                            </div>
                          </Blueprint3.ResizeSensor>
                        </InnerReference>
                      </Reference>
                      <Blueprint3.Overlay
                        autoFocus={false}
                        backdropClassName="bp3-popover-backdrop"
                        backdropProps={Object {}}
                        canEscapeKeyClose={false}
                        canOutsideClickClose={false}
                        enforceFocus={false}
                        hasBackdrop={false}
                        isOpen={false}
                        lazy={true}
                        onClose={[Function]}
                        transitionDuration={100}
                        transitionName="bp3-popover"
                        usePortal={true}
                      />
                    </div>
                  </Manager>
                </Blueprint3.Popover>
              </Blueprint3.Tooltip>
            </TooltipWrapper>
          </Tooltip>
        </div>
      </WorkflowTableStage>
      <WorkflowRowLabel
        isActivityRow={false}
        label="AL1"
      >
        <div
          className="workflow-table-label"
          key="AL1"
        >
          <div>
            AL1
          </div>
        </div>
      </WorkflowRowLabel>
    </div>
  </WorkflowRow>
</Provider>
`;

exports[`Workflow Row shallow mounts 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <WorkflowRow
    stageIntervals={
      Array [
        Object {
          "activityIntervals": Array [
            Object {
              "activeAnalysts": Array [
                "larry",
                "moe",
                "curly",
              ],
              "comment": "interval example",
              "endTime": 1622057187,
              "modificationTime": 1622057667,
              "name": "Event Review",
              "percentAvailable": 100,
              "processingEndTime": 1622057547,
              "processingStartTime": 1622057487,
              "stageName": "AL1",
              "startTime": 1622053587,
              "status": "IN_PROGRESS",
              "storageTime": 1622057607,
            },
          ],
          "comment": "interval example",
          "endTime": 1622057187,
          "modificationTime": 1622057667,
          "name": "AL1",
          "percentAvailable": 100,
          "processingEndTime": 1622057547,
          "processingStartTime": 1622057487,
          "stageMetrics": Object {
            "associatedSignalDetectionCount": 34,
            "eventCount": 21,
            "maxMagnitude": 8,
            "unassociatedSignalDetectionCount": 55,
          },
          "stageMode": "INTERACTIVE",
          "startTime": 1622053587,
          "status": "IN_PROGRESS",
          "storageTime": 1622057607,
        },
      ]
    }
    subRowNames={Array []}
  />
</ContextProvider>
`;
